integration/tests/basic/rpc/test_rpc_estimate_gas.py

```diff
@@ -0,0 +1,173 @@
+        response = self.proxy_api.send_rpc(method="eth_estimateGas", params=transaction)
+        assert "error" not in response
+        assert rpc_checks.is_hex(
+            response["result"]
+        ), f"Invalid response result, `{response['result']}`"
+        transaction["gas"] = int(response["result"], 16)
+        # Deploy contract
+        signed_tx = self.web3_client.eth.account.sign_transaction(
+            transaction, self.account.key
+        )
+        tx = self.web3_client.eth.send_raw_transaction(signed_tx.rawTransaction)
+        contract_deploy_tx = self.web3_client.eth.wait_for_transaction_receipt(tx)
+        return self.web3_client.eth.contract(
+            address=contract_deploy_tx["contractAddress"], abi=contract_interface["abi"]
+        )
+
+    def test_eth_estimate_gas_with_big_int(
+            self, deploy_big_gas_requirements_contract: tp.Any
+    ) -> None:
+        """Check eth_estimateGas request on contracts with big int"""
+        big_gas_contract = deploy_big_gas_requirements_contract
+        trx_big_gas = (
+            big_gas_contract.functions.checkBigGasRequirements().build_transaction(
+                {
+                    "chainId": self.web3_client._chain_id,
+                    "from": self.account.address,
+                    "nonce": self.web3_client.eth.get_transaction_count(
+                        self.account.address
+                    ),
+                    "gas": "0x0",
+                    "gasPrice": hex(self.web3_client.gas_price()),
+                    "value": "0x0",
+                }
+            )
+        )
+        # Check Base contract eth_estimateGas
+        response = self.proxy_api.send_rpc(method="eth_estimateGas", params=trx_big_gas)
+        assert "error" not in response
+        estimated_gas = int(response["result"], 16)
+        assert rpc_checks.is_hex(
+            response["result"]
+        ), f"Invalid response result, `{response['result']}`"
+        trx_big_gas["gas"] = estimated_gas
+        signed_trx_big_gas = self.web3_client.eth.account.sign_transaction(
+            trx_big_gas, self.account.key
+        )
+        raw_trx_big_gas = self.web3_client.eth.send_raw_transaction(
+            signed_trx_big_gas.rawTransaction
+        )
+        deploy_trx_big_gas = self.web3_client.eth.wait_for_transaction_receipt(
+            raw_trx_big_gas
+        )
+        assert deploy_trx_big_gas.get(
+            "status"
+        ), f"Transaction is incomplete: {deploy_trx_big_gas}"
+        assert estimated_gas >= int(
+            deploy_trx_big_gas["gasUsed"]
+        ), "Estimated Gas < Used Gas"
+
+    def test_rpc_estimate_gas_send_neon(self):
+        tx_receipt = self.send_neon(self.sender_account, self.recipient_account, amount=0.001)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
```

Comment:
rename estimate_gas to estimated_gas or just remove it because it used in one place only
```
assert transaction["gas"]  == 30_000
```

Reply:
Yes, it's used in one place, but as `transaction["gas"]` has no any estimation word and this test checks gas estimation, it's just nice to see that here we get that estimation and assert it.

Reply:
Renamed the variable to `estimated_gas`